{"url":"https://api.github.com/repos/willmcgugan/rich/issues/184","repository_url":"https://api.github.com/repos/willmcgugan/rich","labels_url":"https://api.github.com/repos/willmcgugan/rich/issues/184/labels{/name}","comments_url":"https://api.github.com/repos/willmcgugan/rich/issues/184/comments","events_url":"https://api.github.com/repos/willmcgugan/rich/issues/184/events","html_url":"https://github.com/willmcgugan/rich/issues/184","id":666009773,"node_id":"MDU6SXNzdWU2NjYwMDk3NzM=","number":184,"title":"[BUG] Fatal error in rich-4.1.0 library. In version 3.0.3, the error is not critical.","user":{"login":"snooppr","id":61022210,"node_id":"MDQ6VXNlcjYxMDIyMjEw","avatar_url":"https://avatars3.githubusercontent.com/u/61022210?v=4","gravatar_id":"","url":"https://api.github.com/users/snooppr","html_url":"https://github.com/snooppr","followers_url":"https://api.github.com/users/snooppr/followers","following_url":"https://api.github.com/users/snooppr/following{/other_user}","gists_url":"https://api.github.com/users/snooppr/gists{/gist_id}","starred_url":"https://api.github.com/users/snooppr/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/snooppr/subscriptions","organizations_url":"https://api.github.com/users/snooppr/orgs","repos_url":"https://api.github.com/users/snooppr/repos","events_url":"https://api.github.com/users/snooppr/events{/privacy}","received_events_url":"https://api.github.com/users/snooppr/received_events","type":"User","site_admin":false},"labels":[{"id":2200581776,"node_id":"MDU6TGFiZWwyMjAwNTgxNzc2","url":"https://api.github.com/repos/willmcgugan/rich/labels/Needs%20triage","name":"Needs triage","color":"bfdadc","default":false,"description":""}],"state":"closed","locked":false,"assignee":null,"assignees":[],"milestone":null,"comments":3,"created_at":"2020-07-27T06:17:36Z","updated_at":"2020-07-27T20:22:19Z","closed_at":"2020-07-27T20:22:18Z","author_association":"NONE","active_lock_reason":null,"body":"1)  rich-4.1.0 library\r\nhttps://github.com/snooppr/snoop/blob/master/snoop.py#L394\r\n```\r\nIn my code I use rich library (manual update), new version crashes.\r\n    for social_network, net_info in progress1.track(site_data.items(), description=\"\"):\r\n  File \"/home/snoop/.local/lib/python3.7/site-packages/rich/progress.py\", line 710, in track\r\n    progress.refresh()\r\nNameError: name 'progress' is not defined\r\nError in atexit._run_exitfuncs:\r\nTraceback (most recent call last):\r\n```\r\n2) rich-3.0.3 library #I rolled back to this version of the library, but it also has a bug:\r\nLoss of the task completion percentage and loss of bar status (indication line), as well as incorrectly colored lines.\r\n\r\nI rolled back to the stable version of the rich == 1.2.3 library which has no problems.\r\nhttps://github.com/snooppr/snoop/issues/33\r\n\r\n\r\n","closed_by":{"login":"willmcgugan","id":554369,"node_id":"MDQ6VXNlcjU1NDM2OQ==","avatar_url":"https://avatars3.githubusercontent.com/u/554369?v=4","gravatar_id":"","url":"https://api.github.com/users/willmcgugan","html_url":"https://github.com/willmcgugan","followers_url":"https://api.github.com/users/willmcgugan/followers","following_url":"https://api.github.com/users/willmcgugan/following{/other_user}","gists_url":"https://api.github.com/users/willmcgugan/gists{/gist_id}","starred_url":"https://api.github.com/users/willmcgugan/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/willmcgugan/subscriptions","organizations_url":"https://api.github.com/users/willmcgugan/orgs","repos_url":"https://api.github.com/users/willmcgugan/repos","events_url":"https://api.github.com/users/willmcgugan/events{/privacy}","received_events_url":"https://api.github.com/users/willmcgugan/received_events","type":"User","site_admin":false},"performed_via_github_app":null}